File: 13-04-2026/fizzbuzz_validator.py
def is_fizz_buzz(arr):
    foundDigitIndex = next((i for i, x in enumerate(arr) if isinstance(x, int)), -1)
    digit, resultArray = arr[foundDigitIndex], [];
    firstDigit = digit - foundDigitIndex;
    numArray = list(range(firstDigit, firstDigit + len(arr)))

    for i in range(len(numArray)):
        currentElem = numArray[i];
        elementIsFizz, elementIsBuzz = currentElem % 3 == 0, currentElem % 5 == 0

        if elementIsFizz and elementIsBuzz:
            resultArray.append("FizzBuzz")
        elif elementIsFizz:
            resultArray.append("Fizz")
        elif elementIsBuzz:
            resultArray.append("Buzz")
        else:
            resultArray.append(currentElem)
        
        if arr[i] != resultArray[i]:
            return False

    return True


print(is_fizz_buzz([1, 2, "Fizz", 4, "Buzz"])) # True.
print(is_fizz_buzz([13, 14, "FizzBuzz", 16, 17])) # True.
print(is_fizz_buzz([1, 2, "Fizz", 4, 5])) # False.
print(is_fizz_buzz(["FizzBuzz", 16, 17, "Fizz", 19, "Buzz"])) # True.
print(is_fizz_buzz([1, 2, "Fizz", "Buzz", 5])) # False.
print(is_fizz_buzz([97, 98, "Buzz", "Fizz", 101, "Fizz", 103])) # False.
print(is_fizz_buzz(["Fizz", "Buzz", 101, "Fizz", 103, 104, "FizzBuzz"])) # True.

File: 13-04-2026/fizzbuzz_validator.js
function isFizzBuzz(arr) {
  const foundDigitIndex = arr.findIndex((elem) => typeof elem === "number");
  const digit = arr[foundDigitIndex];
  const firstDigit = digit - foundDigitIndex;
  const numArray = Array.from({ length: arr.length }, (_, i) => firstDigit + i);

  for (let i = 0; i < numArray.length; i++) {
    const currentElem = numArray[i];
    const [elementIsFizz, elementIsBuzz] = [
      currentElem % 3 === 0,
      currentElem % 5 === 0,
    ];
    numArray.splice(
      i,
      1,
      elementIsFizz && elementIsBuzz
        ? "FizzBuzz"
        : elementIsFizz
          ? "Fizz"
          : elementIsBuzz
            ? "Buzz"
            : currentElem,
    );
  }

  return arr.join(",").trim() === numArray.join(",").trim();
}

console.log(isFizzBuzz([1, 2, "Fizz", 4, "Buzz"])); // true.
console.log(isFizzBuzz([13, 14, "FizzBuzz", 16, 17])); // true.
console.log(isFizzBuzz([1, 2, "Fizz", 4, 5])); // false.
console.log(isFizzBuzz(["FizzBuzz", 16, 17, "Fizz", 19, "Buzz"])); // true.
console.log(isFizzBuzz([1, 2, "Fizz", "Buzz", 5])); // false.
console.log(isFizzBuzz([97, 98, "Buzz", "Fizz", 101, "Fizz", 103])); // false.
console.log(isFizzBuzz(["Fizz", "Buzz", 101, "Fizz", 103, 104, "FizzBuzz"])); // true.
